brave/brave-core
# Fixes maximum daily ads at 21 instead of 20 (follow up to #3849)

Resolves https://github.com/brave/brave-browser/issues/4207

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:
Regarding the original 4207 issue - see that issue for testing.

This PR refactors the core logic of the 'Should an Ad be shown' logic - as such, QA should check all functionality in this area. 

For Sanity testing purposes...
- Capture the JSON from https://ads-serve.bravesoftware.com/v1/catalog
- Edit it to contain only one campaign, with at least 21 unique ads.
- Ensure that 'segments' 'name' is "technology & computing"
- Alter 'perDay' to 20
- Alter 'dailyCap' to 30
- Save this JSON
- Using Charles Proxy (or similar) use 'Map Local' to redirect the above URL to your edited JSON
- Clean install Brave
- Enable Rewards
- Visit https://brave.com/
- Ensure 20 ads are shown during a given day (and not 21) 

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
